packages/ns-build/server.ts

```diff
@@ -46,15 +46,25 @@ const getProps = async (event: any) => {
    * extracts the `getServerSideProps` function from that module to load
    * the server-side rendering logic dynamically based on the requested URL path.
    */
-  const { getServerSideProps } = require(path)
+  let getServerSidePropsVal
+  try {
+    const { getServerSideProps } = require(path)
+    getServerSidePropsVal = getServerSideProps
+  } catch (err) {
+    showDebugLogs && console.log({ path, err })
+    return {
+      statusCode: 404,
+      body: 'resource not found',
+    }
+  }
```

Comment:
That's a good point, but I'm worried about the code structure
Do you mind restructuring it to something like:
`
const getProps = async (event: any) => {
  try { 
    .....
    const { getServerSideProps } = require(path)
    .....
  } catch (err) {
     showDebugLogs && console.log({ path, err })
     return {
       statusCode: 404,
       body: 'resource not found',
     }
`